### PR TITLE
Add supplier ressource

### PR DESCRIPTION
**Pull request by Berardpi**
https://developers.vendhq.com/documentation/api/0.x/suppliers.html

This endpoint differ from the others as its singular, but it's still returns something the records under a pluralized key:
```
{
  "suppliers": []
}
```

### CODE REVIEW

File: spec/integration/supplier_spec.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  let(:expected_attributes) do
+    {
+      'id' => '6cb5c88c-3d5f-11e0-8697-4040f540b50a',
+      'name'   => 'Supplier Ben'
+    }
+  end
+
+  let(:expected_collection_length) { 1 }
+
+  it_behaves_like "a resource with a collection GET endpoint"
```
Comment (malclocke) [inline spec/integration/supplier_spec.rb]: This spec fails for me, maybe you forgot to add your `suppliers.json` file?
    1) Vend::Resource::Supplier behaves like a resource with a collection GET endpoint gets the collection
         Failure/Error: get_mock_response("#{file_path}.json")
         Errno::ENOENT:
           No such file or directory @ rb_sysopen - ../vend-ruby/spec/mock_responses/suppliers.json

**Comment by malclocke** [comment]
@Berardpi this looks good if you can add the missing mock response file and squash the commits down to 1?

**Comment by Berardpi** [comment]
@malclocke All done. I didn't manage to run the spec on my side so you might want do double check that they're passing

**Comment by malclocke** [comment]
Specs are still failing for me:

```
Failures:

  1) Vend::Resource::Supplier behaves like a resource with a collection GET endpoint gets the collection
     Failure/Error: expect(collection.count).to eq(expected_collection_length)
     WebMock::NetConnectNotAllowedError:
       Real HTTP connections are disabled. Unregistered request: GET https://foo:bar@baz.vendhq.com/api/supplier with headers {'Accept'=>'*/*', 'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3', 'User-Agent'=>'Ruby'}
       
       You can stub this request with the following snippet:
       
       stub_request(:get, "https://foo:bar@baz.vendhq.com/api/supplier").
         with(:headers => {'Accept'=>'*/*', 'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3', 'User-Agent'=>'Ruby'}).
         to_return(:status => 200, :body => "", :headers => {})
       
       registered request stubs:
       
       stub_request(:get, "https://foo:bar@baz.vendhq.com/api/suppliers")
       
       ============================================================
     Shared Example Group: "a resource with a collection GET endpoint" called from ./spec/integration/supplier_spec.rb:13
     # ./lib/vend/http_client.rb:82:in `request'
     # ./lib/vend/resource_collection.rb:125:in `get_next_page'
     # ./lib/vend/resource_collection.rb:33:in `each'
     # ./spec/support/shared_examples/integration.rb:41:in `count'
     # ./spec/support/shared_examples/integration.rb:41:in `block (2 levels) in <top (required)>'
```

**Comment by Berardpi** [comment]
@malclocke Should be green now

**Comment by malclocke** [comment]
@Berardpi thanks.  The changes to the specs broke all of the integration specs, so I have applied a different fix and merged the rest of the changes.  Thank you for the feature, I will roll a new release 0.0.9 with the new endpoint included.

**Pull request closed without merging**